automation/simulator/deck.py

```diff
@@ -0,0 +1,189 @@
+# Intial draft adapted from https://github.com/wynand1004/Projects by @TokyoEdtech
+
+import random
+from ..utils.logger import logger
+
+
+class Card(object):
+    """Card class
+
+    Example use:
+        from automation.simulaiton.deck import Card
+        Card("SA") OR Card("S","A")
+
+    Attributes:
+        suit (str): one of [D,C,H,S,R,B] for Diamond, Club, Heart, Spade, Red, Black
+        val (str): one of [A, K, Q, J, T] for Ace to Ten. Or 2 to 9.
+    """
+
+    def __init__(self, suit, val=None):
+        if len(suit) == 2:
+            self.suit = suit[0].upper()
+            self.val = suit[1].upper()
+        else:
+            self.suit = suit.upper()
+            self.val = val.upper()
+        self._val_to_num = {  # A:1, 2:2, ... T:10
+            "A": 1,
+            "2": 2,
+            "3": 3,
+            "4": 4,
+            "5": 5,
+            "6": 6,
+            "7": 7,
+            "8": 8,
+            "9": 9,
+            "T": 10,
+            "J": 11,
+            "Q": 12,
+            "K": 0,
```

Comment:
What's the reason K = 0 rather than 14?

Reply:
This is makes it easy to then use the modulo operator to wrap around.

```python
12 + 1 % 13 == 0 # king
12 + 2 % 13 == 1 # ace
```